lib/super_digit.rb

```diff
@@ -1,15 +1,60 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(log k*n), but it really feels like O(log n):
+#                              # of times you sum the digits is a operation dependent on the number of
+#                              digits (n) in the input parameter number (which you continuously divide by 10). as for how many times super_digit is called...
+#                              for a k digit number, assume the max possible num is a k-digit number of only 9s.
+#                              The number of super_digit calls:
+#                              k = 1: 1 call for super_digit(9)
+#                              k = 2: 2 calls for super_digit(18) = super_digit(9)
+#                              k = 2: 3 calls for super_digit(99) = super_digit(18) = super_digit (9)
+#                              k = 11: 4 calls for super_digit (99999999999) = super digit (99) = ...
+#                              k = 11111111111: 5 calls for super_digit(a number with k 9s) = super_digit (99999999999)...
+#                              essentially, it looks like an incredibly slow logarithmic growth of k recursive calls
+#                              that would technically result in O(log k* log n) = O(log k*n), but k is so negligible it
+#                              might as well be O(n)... I also don't have space to type 11111111111 1s or 9s...
+#
+# Space Complexity - O(log k) (but again, basically O(1)) based on what was discussed in time complexity. You are creating 2 new variables and a new
+#                    layer on the stack for every recursive call, but as shown above it grows so slowly (and only grows even more slowly)
+#                    that I would argue that it is close to O(1).
```

Comment:
I would say it's O(k) (where k is the number of digits) because you're at-most going to make a k recursive calls.

For something like:  999, the sum will be 27, and then 9, so worst-case losing a digit, or close to 1 digit with each recursive call.

Reply:
OHHHH. This makes sense now -- I was looking at the number itself and not the number of digits.